addons/product/wizard/product_get_pic_using_barcode.py

```diff
@@ -0,0 +1,95 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+import json
+import requests
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class ProductGetPicUsingBarcode(models.TransientModel):
+    """ Get product pictures using barcode on Google Image
+
+    :raises UserError: If the user didn't enter API Key or Search Engine ID in the settings
+    :raises UserError: If there isn't any product without a picture and with a barcode selected.
+    """
+    _name = 'product.get.pic.barcode'
+    _description = 'Get Picture with Barcode Number'
+
+    @api.model
+    def default_get(self, fields):
+        default = super().default_get(fields)
+
+        google_custom_search_cx = self.env['ir.config_parameter'] \
+            .get_param('google.custom_search.cx')
+        google_custom_search_key = self.env['ir.config_parameter'] \
+            .get_param('google.custom_search.key')
+
+        if not google_custom_search_cx or not google_custom_search_key:
+            raise UserError(_("You didn't configure properly your API Key or Search Engine ID."))
+
+        if self._context.get('active_model') == 'product.template':
+            product_ids = self.env['product.template'].browse(self._context.get('active_ids')) \
+                .product_variant_ids.filtered(lambda p: not p.image_1920 and p.barcode)
+        else:
+            product_ids = self.env['product.product'].browse(self._context.get('active_ids')) \
+                .filtered(lambda p: not p.image_1920 and p.barcode).ids
+
+        if len(product_ids) == 0:
+            raise UserError(_("There isn't any product without a picture and with a barcode."))
+
+        default.update({
+            'google_custom_search_cx': google_custom_search_cx.strip(),
+            'google_custom_search_key': google_custom_search_key.strip(),
```

Comment:
(sorry, this comment was lost in the review) By doing this you're leaking the value of those secrets into the wizard fields, which anyone can read. And anyway you only need a boolean field to know whether the warning should be displayed. Please do that instead. Use the sys params directly when needed but don't return them or store them anywhere.